### PR TITLE
Build popup window and Section 2 using javaScript

### PR DESCRIPTION
**Pull request by Jord4N-AC**
For this milestone of my portfolio, I have implemented a popup window with the details of desktop and mobile Cards, I also built Section 2 Dinamiclly to do this I stored the information in a JavaScript object, I did this pair-programming with mi partner @kemigabocatherine, We implemented this in both projects:

### Features
- When the user clicks the button, the popup with details appears
- When the user clicks the close (X) button, the popup disappears
- Make popup window responsive for mobile and desktop sizes
- When pressing the Esc key or clicking out of the popup, it also disappears (**extra**)
- When resizing the screen the content on Section 2 and popups updates according to the Figma template (**extra**)

### Requirements
- Use basic JavaScript syntax
- Use JavaScript to manipulate DOM elements
- Use JavaScript events
- Understand how to parse a Figma design to create a UI
- Use objects to store and access data

### General Requirements
- Check that there are no linter errors
- Use correct GitHub flow
- Document my work in a professional way
- Follow best practices for HTML, CSS & JavaScript
- Open a pull request

### CODE REVIEW

File: main.js [hunk 1]
```diff
@@ -28,7 +28,345 @@ function disapearEcsAndResize(event) {
...
+      icon: [
+        'images/Icons/see_live_icon.svg',
+        'images/Icons/see_source_icon.svg',
+      ],
+      links: [
+        'https://github.com/Jord4N-AC/portfolio',
+      ],
+    },
+  },
+};
```
Comment (medaminedev66) [inline main.js:186]: - [ ] The pop-up works as expected. great job 👍 . I would recommend changing the structure of your array by using objects to separate the projects like the following example:
```js
const projects = [{
    name: "Submarine game",
    description: "A Phaser3 platform game built as part of the Microverse JavaScript program.",
    image: "link_to_image.jpg",
    technologies: ["Phaser3", "JavaScript", "Jest"],
    link: "link_to_live_version.com",
},
{
    name: "Second project",
    description: "A Phaser3 platform game built as part of the Microverse JavaScript program.",
    image: "link_to_image.jpg",
    technologies: ["Phaser3", "JavaScript", "Jest"],
    link: "link_to_live_version.com",
}]
```
By implementing this structure, you will be able to add your future projects easily without any confusion.

**Review by medaminedev66 — CHANGES_REQUESTED** [review]
Hi @Jord4N-AC and @kemigabocatherine , :wave:

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight
- ✔️Your design is amazing and matches the Figma design
- ✔️You implemented the `pop-up` in a good way
- ✔️You have a descriptive PR

## Required Changes ♻️

_See the inline comments below for more required changes_

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: main.js (hunk 1 above)
Comment (Jord4N-AC) [reply main.js:186]: I don't really understand this

Comment (Kingstalux) [reply main.js]: not resolved

Comment (Jord4N-AC) [reply main.js]: You mean to my new changes?

File: main.js [hunk 2]
```diff
@@ -38,141 +38,31 @@ menuLinks.forEach((link) => {
...
-      ],
+const cardsContent = [
+  // Content in some array follows this order
+  // [mobile, mobilePopup, desktop, desktopPopup]
+  // ('') empty values mean there no content for that Part
+  cardsContent[ind][header][mobile]
+  {
+    header: [
+      'Multi-Post Stories', 'Multi Post Stories',
+      'Multi-Post Stories', 'Keeping track of hundreds  of components website',
```
Comment (Kingstalux) [inline main.js:48]: - [ ] This `cardsContent` should be an array of objects which store objects containing the information for each project. Each of the objects should contain at least the following information about the project: name, description, featured image, technologies, link to live version, and link to the project source code. What you've done here is grouping all the `headers` together, `descriptions` together and so on which should not be the case.
- Your array should look like this:
Annotation 2022-03-05 065014
- Write all the information about one project inside a single object before opening another object and writing the information about the next project 👍

**Review by Kingstalux — CHANGES_REQUESTED** [review]
Hi @Jord4N-AC and @kemigabocatherine , 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

149637855-7722d04c-b45d-4185-925f-7de047106461

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: main.js (hunk 1 above)
Comment (Kingstalux) [reply main.js]: - You are supposed to place all the project info inside a single object, if you have 7 projects, you should have 7 objects inside your array but your objects have been grouped according to heading, description, and so on.
- Your project
Annotation 2022-03-05 065544

File: main.js (hunk 2 above)
Comment (Jord4N-AC) [reply main.js:48]: OMG This is not possible, I haven't grouped headers and descriptions together, if you didn't realize in the comment //[mobile, mobilePopup, desktop, desktopPopup]
says that arrays that have four values, each one belongs to different content, the origin design changes some headers paragraphs and images when in mobile, desktop and when mobilePopup, desktopPopup

Comment (Kingstalux) [reply main.js:48]: Hello @Jord4N-AC are you available for a zoom call so that you can better explain yourself?

File: main.js [hunk 3]
```diff
@@ -34,3 +34,303 @@ window.addEventListener('click', disapearEcsResizeClickout);
 menuLinks.forEach((link) => {
   link.addEventListener('click', disapear);
 });
+
+// Section 2 & PopUp Window
+
+const cardsContent = [
```
Comment (mateo951) [inline main.js:40]: Well done 🚀🔥 The JavaScript code implements an array of objects to store JavaScript objects containing the information for each project.

File: main.html [hunk 4]
```diff
@@ -69,118 +69,7 @@ <h3 class="intro-header2">I’am a Software Developer</h3>
...
-                </p>
-                <ul class="tag-group">
-                    <li class="tag">html</li>
-                    <li class="tag">bootstrap</li>
-                    <li class="tag">Ruby</li>
-                </ul>
-            </div>
-            <a class="btn btn-2 see-project-btn" href="#">See Project</a>
-        </div>
+        <!-- This Section was built dinamically in javaScript  -->
```
Comment (mateo951) [inline main.html:72]: Wonderfull work 🤓 The projects section is created dynamically with the information from the projects array as expected  🔥

File: main.js [hunk 5]
```diff
@@ -34,3 +34,303 @@ window.addEventListener('click', disapearEcsResizeClickout);
...
+    if (event.target === modalWindow) {
+      closeModal();
+    }
+  });
+}
+
+buttons.forEach((btn, ind) => {
+  btn.addEventListener('click', () => {
+    popupInd = ind;
+    showModalWindow(ind);
```
Comment (mateo951) [inline main.js:303]: The popup with details about the project appears and closes perfectly  😎

File: style.css [hunk 6]
```diff
@@ -4,7 +4,6 @@ html {
```
Comment (mateo951) [inline style.css:4]: The popup is resizing and showing properly on mobile and desktop sizes 🤯 The effects are looking nice💣

**Review by mateo951 — APPROVED** [review]
### Hi Team, 

👌 **Your project highlights** 😎 
- General Requirements 🙌
- App JS functionality is implemented beautifully 💪😎

Your project is complete!  There is nothing else to say other than... it's time to merge it ✔
### Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Jord4N-AC** [comment]
Thanks a lot to all reviewers for the feedback

**Pull request merged**